seaborn/categorical.py

```diff
@@ -1129,15 +1168,228 @@ def plot(self, ax, kws):
             ax.invert_yaxis()
 
 
-class _SwarmPlotter(_BoxPlotter):
+class _SwarmPlotter(_CategoricalScatterPlotter):
 
-    def __init__(self):
+    def __init__(self, x, y, hue, data, order, hue_order,
+                 split, orient, color, palette):
+        """Initialize the plotter."""
+        self.establish_variables(x, y, hue, data, orient, order, hue_order)
+        self.establish_colors(color, palette, 1)
 
-        pass
+        # Set object attributes
+        self.split = split
+        self.width = .8
 
-    def plot(self, ax):
+    def overlap(self, xy_i, xy_j, d):
+        """Return True if two circles with the same diameter will overlap."""
+        x_i, y_i = xy_i
+        x_j, y_j = xy_j
+        return np.linalg.norm([x_i - x_j, y_i - y_j]) < d
```

Comment:
`(x_i - x_j)**2 + (y_i - y_j)**2 < d**2` will likely be much faster, `np.linalg.norm` seem overkill for a norm2 comparison. Unless I misunderstand the usage `np.linalg.norm`. Though you might be able, if you use numpy to vectorize `prune_candidates`, which might be faster.